manim/mobject/polyhedra.py

```diff
@@ -0,0 +1,360 @@
+
+    def update_faces(self, m):
+        face_coords = self.extract_face_coords()
+        new_faces = self.create_faces(face_coords)
+        self.faces.match_points(new_faces)
+
+    def extract_face_coords(self):
+        """Extracts the coordinates of the vertices in the graph.
+        Used for updating faces.
+        """
+        new_vertex_coords = []
+        for v in self.graph.vertices:
+            new_vertex_coords.append(self.graph[v].get_center())
+        layout = dict(enumerate(new_vertex_coords))
+        return [[layout[j] for j in i] for i in self.faces_list]
+
+
+class Tetrahedron(Polyhedron):
+    """A tetrahedron, one of the five platonic solids.
+
+    Parameters
+    ----------
+    edge_length:
+        The length of an edge between any two vertices.
+
+    Examples
+    --------
+
+    .. manim:: TetrahedronScene
+        :save_last_frame:
+
+        class TetrahedronScene(ThreeDScene):
+            def construct(self):
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                obj = Tetrahedron()
+                self.add(obj)
+    """
+
+    def __init__(self, edge_length: float = 1, **kwargs):
+        unit = edge_length * np.sqrt(2) / 4
+        Polyhedron.__init__(
+            self,
+            vertex_coords=[
+                np.array([unit, unit, unit]),
+                np.array([unit, -unit, -unit]),
+                np.array([-unit, unit, -unit]),
+                np.array([-unit, -unit, unit]),
+            ],
+            faces_list=[[0, 1, 2], [3, 0, 2], [0, 1, 3], [3, 1, 2]],
+            **kwargs
+        )
+
+
+class Octahedron(Polyhedron):
+    """An octahedron, one of the five platonic solids.
+
+    Parameters
+    ----------
+    edge_length:
+        The length of an edge between any two vertices.
+
+    Examples
+    --------
+
+    .. manim:: OctahedronScene
+        :save_last_frame:
+
+        class OctahedronScene(ThreeDScene):
+            def construct(self):
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                obj = Octahedron()
+                self.add(obj)
+    """
+
+    def __init__(self, edge_length: float = 1, **kwargs):
+        unit = edge_length * np.sqrt(2) / 2
+        Polyhedron.__init__(
+            self,
+            vertex_coords=[
+                np.array([unit, 0, 0]),
+                np.array([-unit, 0, 0]),
+                np.array([0, unit, 0]),
+                np.array([0, -unit, 0]),
+                np.array([0, 0, unit]),
+                np.array([0, 0, -unit]),
+            ],
+            faces_list=[
+                [2, 4, 1],
+                [0, 4, 2],
+                [4, 3, 0],
+                [1, 3, 4],
+                [3, 5, 0],
+                [1, 5, 3],
+                [2, 5, 1],
+                [0, 5, 2],
+            ],
+            **kwargs
+        )
+
+
+class Icosahedron(Polyhedron):
+    """An icosahedron, one of the five platonic solids.
```

Comment:
It might be useful to note how many sides the Polyhedron has.
```suggestion
    """An icosahedron, one of the platonic solids, and one with twenty faces.
```
or
```suggestion
    """An icosahedron, the polyhedron with twenty faces.
```

Reply:
Comment above.